Getting the HumHub login page as document content

Hi everyone, 

I have a weird issue: 

1. HumHub installed classically with nginx
2. onlyoffice-ie in docker container on different host
3. JWT secret configured 
4. both behind reverse proxy which does the SSL termination with valid certificates

When I open a document in a space, it does not show the content but the text content of the humhub login page:
![image](https://user-images.githubusercontent.com/5879453/89099408-3bac1800-d3ef-11ea-8518-5e4c8c988aed.png)

I've turned up the logging in the onlyoffice container:
```
[2020-08-01T09:37:57.745] [DEBUG] nodeJS - getCallbackByUserIndex: docId = 33844eda177ee07d6b58 userIndex = 1 callbacks = ^E{"userIndex":1,"callback":"https://<humhub.server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=33844eda177ee07d6b58"}
[2020-08-01T09:37:57.746] [DEBUG] nodeJS - postData request: docId = 33844eda177ee07d6b58;url = https://<humhub.server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrac
k&key=33844eda177ee07d6b58;data = {"key":"33844eda177ee07d6b58","status":1,"users":["244a1fe1-0f52-450a-8e70-21598427d110"],"actions":[{"type":1,"userid":"244a1fe1-0f52-450a-8e7
0-21598427d110"}]}
[2020-08-01T09:37:57.809] [ERROR] nodeJS - postData error: docId = 33844eda177ee07d6b58;url = https://<humhub.server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&
key=33844eda177ee07d6b58;data = {"key":"33844eda177ee07d6b58","status":1,"users":["244a1fe1-0f52-450a-8e70-21598427d110"],"actions":[{"type":1,"userid":"244a1fe1-0f52-450a-8e70-
21598427d110"}]}
Error: Error response: statusCode:302 ;body:

    at Request._callback (/snapshot/server/build/server/Common/sources/utils.js:0:0)
    at Request.init.self.callback (/snapshot/server/build/server/Common/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:198:13)
    at IncomingMessage.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:286:20)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1143:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
[2020-08-01T09:37:57.812] [DEBUG] nodeJS - sendData: docId = 33844eda177ee07d6b58;type = auth
[2020-08-01T09:37:57.813] [DEBUG] nodeJS - pubsub message start:{"type":8,"docId":"33844eda177ee07d6b58","description":"Error on save server subscription!"}
[2020-08-01T09:37:57.813] [DEBUG] nodeJS - sendData: docId = 33844eda177ee07d6b58;type = warning
[2020-08-01T09:37:57.947] [INFO] nodeJS - GET /doc/fbc4437584f8ff0304c5/c/info 1ms 200
[2020-08-01T09:37:58.078] [INFO] nodeJS - GET /doc/fbc4437584f8ff0304c5/c/812/e0zkfaxj/websocket 0ms (unfinished)
[2020-08-01T09:37:58.080] [DEBUG] nodeJS - sendData: docId = undefined;type = license
[2020-08-01T09:37:58.109] [INFO] nodeJS - data.type = auth id = undefined
```
The onlyoffice server is not authenticated when it is trying to download the document and is 302 redirected to the HumHub login page. What could be the cause of this? I'd be very grateful for any pointers. 
Best regards
Thomas

So, I now get a different error message:
```
[2020-08-03T13:21:19.645] [DEBUG] nodeJS - postData request: docId = bf52e5ff9817b2a26c01;url = https://<humhub-server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=bf52e5ff9817b2a26c01;data = {"key":"bf52e5ff9817b2a26c01","status":1,"users":["1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"],"actions":[{"type":1,"userid":"1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"}]}
[2020-08-03T13:21:19.672] [ERROR] nodeJS - postData error: docId = bf52e5ff9817b2a26c01;url = https://<humhub-server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=bf52e5ff9817b2a26c01;data = {"key":"bf52e5ff9817b2a26c01","status":1,"users":["1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"],"actions":[{"type":1,"userid":"1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"}]}
Error: Error response: statusCode:500 ;body:
{"error":1,"message":"Invalid JWT signature"}
    at Request._callback (/snapshot/server/build/server/Common/sources/utils.js:0:0)
    at Request.init.self.callback (/snapshot/server/build/server/Common/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:198:13)
    at IncomingMessage.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:286:20)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1143:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
[2020-08-03T13:21:19.674] [DEBUG] nodeJS - pubsub message start:{"type":8,"docId":"bf52e5ff9817b2a26c01","description":"Error on save server subscription!"}
[2020-08-03T13:21:19.675] [DEBUG] nodeJS - sendData: docId = bf52e5ff9817b2a26c01;type = warning
[2020-08-03T13:21:19.679] [DEBUG] nodeJS - sendData: docId = bf52e5ff9817b2a26c01;type = authChanges
[2020-08-03T13:21:19.681] [DEBUG] nodeJS - sendData: docId = bf52e5ff9817b2a26c01;type = auth
[2020-08-03T13:21:19.681] [DEBUG] nodeJS - Start command: docId = bf52e5ff9817b2a26c01 {"withAuthorization":true,"c":"open","id":"bf52e5ff9817b2a26c01","userid":"1240ad4b-ad1c-485c-ae60-89fb58bdb9c5","format":"docx","url":"https://<humhub-server>/humhub/index.php?r=onlyoffice%2Fbackend%2Fdownload&key=bf52e5ff9817b2a26c01","title":"testdatei.docx","lcid":7,"nobase64":true}
[2020-08-03T13:21:19.683] [DEBUG] nodeJS - Response command: docId = bf52e5ff9817b2a26c01 {"type":"open","status":"ok","data":{"Editor.bin":"https://<onlyoffice-server>/cache/files/bf52e5ff9817b2a26c01/Editor.bin/Editor.bin?md5=htkieDf85tykjj4TSySoRg&expires=1599052880&disposition=attachment&filename=Editor.bin"}}
```

The JWT secret is correct and the weirdest thing is: The changes to the document are actually saved.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,10 @@
 # Change Log
 
+## 2.1.2
+## Fixed
+- Fixed an issue that prevented plugin from removing (#14)
+
+
 ## 2.1.1
 ## Changed
 - Improved migration failures (#12)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Change Log
 
+## 2.1.3
+## Fixed
+- HumHub 1.6 compatibility (#17, #19)
+
 ## 2.1.2
 ## Fixed
 - Fixed an issue that prevented plugin from removing (#14)

--- a/README.md
+++ b/README.md
@@ -30,7 +30,7 @@ Community Edition vs Integration Edition comparison can be found [here](#onlyoff
 
 ## Installing HumHub ONLYOFFICE integration plugin
 
-Either install it from [HumHub Marketplace](https://www.humhub.com/en/marketplace/onlyoffice/) or simply clone the repository inside one of the folder specified by `moduleAutoloadPaths` parameter. Please see [HumHub Documentation](http://docs.humhub.org/dev-environment.html#external-modules-directory) for more information.
+Either install it from [HumHub Marketplace](https://www.humhub.com/en/marketplace/onlyoffice/) or simply clone the repository inside one of the folder specified by `moduleAutoloadPaths` parameter. Please see [HumHub Documentation](https://docs.humhub.org/docs/develop/environment#module-loader-path) for more information.
 
 ## Configuring HumHub CONLYOFFICE integration plugin
 

--- a/README.md
+++ b/README.md
@@ -30,7 +30,7 @@ Community Edition vs Integration Edition comparison can be found [here](#onlyoff
 
 ## Installing HumHub ONLYOFFICE integration plugin
 
-Either install it from HumHub Marketplace or simply clone the repository inside one of the folder specified by `moduleAutoloadPaths` parameter. Please see [HumHub Documentation](http://docs.humhub.org/dev-environment.html#external-modules-directory) for more information.
+Either install it from [HumHub Marketplace](https://www.humhub.com/en/marketplace/onlyoffice/) or simply clone the repository inside one of the folder specified by `moduleAutoloadPaths` parameter. Please see [HumHub Documentation](http://docs.humhub.org/dev-environment.html#external-modules-directory) for more information.
 
 ## Configuring HumHub CONLYOFFICE integration plugin
 

--- a/controllers/BackendController.php
+++ b/controllers/BackendController.php
@@ -10,15 +10,18 @@ namespace humhub\modules\onlyoffice\controllers;
 
 use Yii;
 use yii\web\HttpException;
-use yii\helpers\Url;
 use humhub\modules\file\models\File;
-use humhub\modules\file\libs\FileHelper;
 use humhub\modules\user\models\User;
 use humhub\components\Controller;
 use \Firebase\JWT\JWT;
 
 class BackendController extends Controller
 {
+    /**
+     * @inheritdoc
+     * Allow server to server configuration without any authentication
+     */
+    public $access = \humhub\components\access\ControllerAccess::class;
 
     /**
      * @var File
@@ -91,7 +94,7 @@ class BackendController extends Controller
 
                 try {
                     $ds = JWT::decode($token, $module->getJwtSecret(), array('HS256'));
-                    $data = (array) $ds->payload;
+                    $data = (array)$ds->payload;
                 } catch (\Exception $ex) {
                     throw new \Exception('Invalid JWT signature');
                 }
@@ -133,7 +136,7 @@ class BackendController extends Controller
                         throw new \Exception('Could not save onlyoffice document: ' . $data["url"]);
                     }
 
-                break;
+                    break;
             }
 
         } catch (\Exception $e) {

--- a/controllers/OpenController.php
+++ b/controllers/OpenController.php
@@ -16,6 +16,11 @@ use humhub\modules\onlyoffice\components\BaseFileController;
 
 class OpenController extends BaseFileController
 {
+    /**
+     * @inheritdoc
+     * Allow access to this controller without any authentication (guest access)
+     */
+    public $access = \humhub\components\access\ControllerAccess::class;
 
     /**
      * Opens the document in modal

--- a/module.json
+++ b/module.json
@@ -5,7 +5,7 @@
     "keywords": [
         "office", "word", "excel", "presentation", "document collaboration", "online editors", "edit documents online", "edit word online", "edit presentation online"
     ],
-    "version": "2.1.1",
+    "version": "2.1.2",
     "humhub": {
         "minVersion": "1.3"
     },

--- a/module.json
+++ b/module.json
@@ -5,7 +5,7 @@
     "keywords": [
         "office", "word", "excel", "presentation", "document collaboration", "online editors", "edit documents online", "edit word online", "edit presentation online"
     ],
-    "version": "2.1.2",
+    "version": "2.1.3",
     "humhub": {
         "minVersion": "1.3"
     },

